packages/craco/README.md

```diff
@@ -125,13 +129,97 @@ Or build your app:
+>[the CRACO configuration file template](#the-craco-configuration-file-template)_
+
+## Notes on CRA Configurations and Problem Solving  
+Keep in mind that there are _some_ configuration settings available to CRA without CRACO. 
+
+Getting exactly what you want may involve a combination of making changes your CRACO configuration file and by using 
+some of the more limited _but still important_ settings available in Create React App. 
+
+Before jumping into customizing your _CRACO_ configuration, step back and think about each part of the problem you're 
+trying to solve. Be sure to review these resources on the CRA configuration, as it may save you time:
+
+ - [Important Environment Variables that Configure CRA](https://create-react-app.dev/docs/advanced-configuration)
+ - [Learn about using `postbuild` commands in `package.json`](https://stackoverflow.com/a/51818028/4028977)
+ - [Proxying API or other Requests](https://create-react-app.dev/docs/proxying-api-requests-in-development/), or "how 
+ to integrate CRA's dev server with a second backend": 
+ [problem statement](https://github.com/facebook/create-react-app/issues/147)
+ - [Search CRACO issues, for gotchas, hints and examples](https://github.com/gsoft-inc/craco/issues?q=is%3Aissue+sort%3Aupdated-desc)
+
+## A Note on Ejecting CRA
+Avoiding ejecting is a major goal for many CRACO users. However, if you're still learning toolchains and modern 
+frontend workflows, it may be helpful to create a sample ejected CRA project to see how the default CRA app configures 
+each of the components. 
+
+While CRACO's sample configuration file inherits directly from CRA's default settings, seeing the default CRA config in
+the ejected CRA file structure may give you useful perspective. 
+
+You may even want to try testing a change in the ejected app to better understand how it would be done with your CRACO 
+config-based project. Okay! On with the CRACO Config!
+
+## The CRACO Configuration File
+CRACO is configured with a single `craco.config.js` file. This file is divided into sections representing the major 
+parts of what makes up the default create react app. 
+
+The sample file below is a starter template you can use to begin configuring your CRA-based project. Each section 
+contains a primary configuration area, `loaderOptions` or `configure`. These config areas are where you will make most 
+of your detailed changes.
+
+### Direct (object literal) Versus Functional Config Definitions
+You, (or perhaps your IDE) may have noticed that the sections have duplicate keys, i.e. loaderOptions is listed twice.
+
+The reason for this is to allow you to choose between object literal or functionally defined configuration choices. 
+There are a few reasons for this:
+
+1. Sometimes it may be faster to test a minor change using keys. 
+1. Other times a functional definition is necessary to get the right configuration.
+1. While not common, a setting may **only** work if you use one or the other! See, 
```

Comment:
Basically the only reason functional definitions are available is to allow someone to dynamically choose a configuration based on a set of variables available during execution.

I am not sure about writing that much documentation for this particular feature?

Reply:
I think when I wrote this, I had actually lost a fair bit of time debugging #172. I had thought that functional definition would get equal treatment to object literals.

Reply:
It seems pretty obvious to me as it is currently showcase in the config overview. That begin said I am the one who wrote it ;)

Hoewever, from the issues I saw, it doesn't seems to bring much confusion.

Maybe add it later?

Reply:
Moved to beginner section.